Feature/search

amiryehu opened the pull request.

Commit:
creating Header and Search folders and files

Commit:
right side icons and avatar

Commit:
wating for icins

Commit:
Merge branch 'develop' of https://github.com/amiryehu/dispatcher into feature/header

Commit:
create Avatar and SettingsIcon & NotificationsIcon

Commit:
create search component
fixing filter border

Commit:
no message

ofekbd-moveo commented on src/Utils/Constances.ts:
```diff
@@ -27,3 +27,8 @@ export const SeparatingLine = styled.div`
     border-color: ${colors.gray};
 `;
+
+export const screenSize = {
```
nice!

ofekbd-moveo commented on src/components/Avatar/Avatar-style.ts:
```diff
@@ -0,0 +1,15 @@
+export const AavatarWrapper = styled.div`
+    background-color: ${colors.blue};
+    color: ${colors.white};
+    font-size: 14px;
```
rem?

ofekbd-moveo commented on src/components/Avatar/Avatar.tsx:
```diff
@@ -0,0 +1,19 @@
+  const {firstName, lastName} = props;
+  return (
+    <AavatarWrapper >
+      {firstName.charAt(0) + lastName.charAt(0)}
```
What if my name is Ofek Ben David or even longer? it should be OBD.
Also, try to make each letter to be a capital letter.

ofekbd-moveo commented on src/components/Filter/Filter-style.ts:
```diff
@@ -1,15 +1,27 @@
+  display: flex;
+  width: 175px;
+  &.searchFilter {
+    width: 145px;
```
It will be better to use % and give min-width to limit the minimal width

amiryehu replied:
I tried to use %, after I connect the component of a filter container I will check again that its ok
* I change it for 'max-width' for now

amiryehu replied:
It makes other problems in search component.
I change it back.

ofekbd-moveo commented on src/components/Header/Settings.tsx:
```diff
@@ -0,0 +1,16 @@
+import * as React from 'react';
+import IconButton from '@mui/material/IconButton';
+import Tooltip from '@mui/material/Tooltip';
+import { ReactComponent as Settings } from "../../assets/Icons/settings.svg";
```
why not "import settings from "../../assets/Icons/settings.svg";
for what { ReactComponent as Settings }?

amiryehu replied:
"import settings from "../../assets/Icons/settings.svg";
this syntax did not work no, this way I import it as a component

ofekbd-moveo commented on src/components/Login/LogIn.tsx:
```diff
@@ -2,7 +2,6 @@ import React from "react";
 import Button from "../Button/Button";
 import {  LogInPage,  LogoComponent,  WelcomeComponent,  WelcomeWrapper,  WelcomeTitle,  WelcomeMessage,} from "./LogIn-style";
 import { ReactComponent as LogInLogo } from "../../assets/Icons/logo.svg";
```
ReactComponent?

ofekbd-moveo replied:
?

ofekbd-moveo commented on src/components/Search/Search-style.ts:
```diff
@@ -0,0 +1,52 @@
+  isFocused: boolean
+}
+
+export const PaperWrapper = styled.div<IPaperWrapperProps>`
```
Use here 'transition' - think which effect it will provide

amiryehu replied:
I saw, I got stuck with the syntax

ofekbd-moveo commented on src/components/Search/Search-style.ts:
```diff
@@ -0,0 +1,52 @@
+  width: 100%;
+  align-items: center;
+  justify-content: flex-end;
+  @media screen and (max-width: ${screenSize.tablet}) {
```
Necessery?

amiryehu replied:
Yes, it hides the filter when the screen is small

ofekbd-moveo commented on src/components/Search/Search.tsx:
```diff
@@ -0,0 +1,60 @@
+import { paperStyle, SearchWrapper, FilterDividerWrapper, PaperWrapper, inputBaseStyle,} from "./Search-style";
+import { useState } from "react";
+
+const SearchForm = (props: any) => {
```
Do NOT use any!!!!! if you do not need props - don't expect to get them.

amiryehu replied:
Will not happen again.

ofekbd-moveo commented on src/components/Search/Search.tsx:
```diff
@@ -0,0 +1,60 @@
+
+const SearchForm = (props: any) => {
+  const [isFocused, setIsFocused] = useState(false);
+  React.useEffect(() => {
```
Necessary?

amiryehu replied:
Deleted

Commit:
responsive ligin page

Commit:
clean search components
and responsive login page

ofekbd-moveo commented on src/components/Search/Search.tsx:
```diff
@@ -0,0 +1,60 @@
+    console.log(isFocused);
+  }, [isFocused]);
+  return (
+    <PaperWrapper isFocused={isFocused}>
```
what are the advantages of PaperWrapper? why not just div?

amiryehu replied:
It's a styled-component, it gets props and that's how I expand the search

ofekbd-moveo commented on src/components/Search/Search.tsx:
```diff
@@ -0,0 +1,60 @@
+  }, [isFocused]);
+  return (
+    <PaperWrapper isFocused={isFocused}>
+      <Paper component="form" sx={{ ...paperStyle }}>
```
Why Paper?

amiryehu replied:
Paper is from mui

ofekbd-moveo commented on src/components/Search/Search.tsx:
```diff
@@ -0,0 +1,60 @@
+      <Paper component="form" sx={{ ...paperStyle }}>
+        <SearchWrapper>
+          <SearchIcon style={{ ...searchIconStyle }} />
+          <InputBase
```
Why did you choose 'InputBase'? Why not TextField for example?

amiryehu replied:
Its from mui too

ofekbd-moveo commented on src/components/Search/Search.tsx:
```diff
@@ -0,0 +1,60 @@
+        <SearchWrapper>
+          <SearchIcon style={{ ...searchIconStyle }} />
+          <InputBase
+            onClick={() => setIsFocused(true)}
```
Input element has focus and blur attributes by default. (like hover)
If it is used for styling - you can remove it.

ofekbd-moveo commented on src/components/Search/Search.tsx:
```diff
@@ -0,0 +1,60 @@
+          <Divider sx={{ ...dividerStyle }} orientation="vertical" />
+          <FilterWrapper className="searchFilter">
+            <BasicSelect
+              title="Eevrithing"
```
*Everything

ofekbd-moveo commented on src/components/Search/Search.tsx:
```diff
@@ -0,0 +1,60 @@
+
+export default SearchForm;
+
+// import React from "react";
```
delete

Commit:
fixing search pr
- avatar names function
- change some of the logos import
- change to textField (Paper tag from mui not change)

Commit:
no message

Commit:
search without Paper

Commit:
app

ofekbd-moveo commented on src/components/Avatar/Avatar.tsx:
```diff
@@ -0,0 +1,29 @@
+
+const getFirtLetterFromNames = (name: string) => {
+  const splitNameArray = name.split(' ');
+  splitNameArray.length = 3;
```
Don't limit the name's length by 3. You can use just array.length.

amiryehu replied:
If there is more than three letters they will be outside the Avatar.

ofekbd-moveo commented on src/components/Avatar/Avatar.tsx:
```diff
@@ -0,0 +1,29 @@
+const getFirtLetterFromNames = (name: string) => {
+  const splitNameArray = name.split(' ');
+  splitNameArray.length = 3;
+  for(let i = 0; i < splitNameArray.length ; i++) {
```
map() will be better

ofekbd-moveo commented on src/components/Header/Settings.tsx:
```diff
@@ -0,0 +1,18 @@
+      <Tooltip title="Coming soon">
+        <IconButton>
+          <img src={Settings} />
+          {/* <Settings/> */}
```
delete

ofekbd-moveo commented on src/components/Search/Search-style.ts:
```diff
@@ -0,0 +1,64 @@
+`;
+
+export const SearchContainer = styled.div`
+  background-color: #fff;
```
${colors.<>}

ofekbd-moveo commented on src/components/Search/Search.tsx:
```diff
@@ -0,0 +1,42 @@
+  
+  return (
+    <SearchContainerWrapper isFocused={isFocused}>
+      <SearchContainer>
```
Maybe this container is redundant. Did you try to pass the 'SearchContainer' style to 'SearchContainerWrapper'?

amiryehu replied:
with 'SearchContainerWrapper' I expand and narrow the search box in focus.
and 'SearchContainer' helps me organize the search and icon on the left side.

ofekbd-moveo commented on src/components/Search/Search.tsx:
```diff
@@ -0,0 +1,42 @@
+
+        <FilterDividerWrapper>
+          <Divider sx={{ ...dividerStyle }} orientation="vertical" />
+          <FilterWrapper className="searchFilter">
```
here too. why so many containers and wrappers?

OrLevi-1 commented on src/components/Avatar/Avatar-style.ts:
```diff
@@ -0,0 +1,15 @@
+import styled from "styled-components";
+import { colors } from "../../Utils/Constances";
+
+export const AavatarWrapper = styled.div`
```
mistype Avatar

OrLevi-1 commented on src/components/Avatar/Avatar.tsx:
```diff
@@ -0,0 +1,29 @@
+  onClick?: ()=>void
+}
+
+const getFirtLetterFromNames = (name: string) => {
```
getFirst*

OrLevi-1 commented on src/components/Search/Search-style.ts:
```diff
@@ -0,0 +1,64 @@
+  }
+`;
+
+export const dividerStyle = {
+  height: 28,
+  m: 0.5,
+};
```
Necessary? what is "m"?

OrLevi-1 commented on src/components/Search/Search-style.ts:
```diff
@@ -0,0 +1,64 @@
+export const SearchContainer = styled.div`
+  background-color: #fff;
+  color: rgba(0, 0, 0, 0.87);
+  box-shadow: 0px 2px 1px -1px rgb(0 0 0 / 20%),
+    0px 1px 1px 0px rgb(0 0 0 / 14%), 0px 1px 3px 0px rgb(0 0 0 / 12%);
```
necessary to use 3 white colors for the box shadow?

OrLevi-1 commented on src/components/Search/Search.stories.tsx:
```diff
@@ -0,0 +1,15 @@
+} as ComponentMeta<typeof SearchForm>;
+
+const Template: ComponentStory<typeof SearchForm> = () => (
+  <SearchForm ></SearchForm>
```
you can use `<SearchForm/>` here

Commit:
fix search pr

Commit:
clean filter and search
